### PR TITLE
Readme cleanup with additional local dev steps

### PR DESCRIPTION
**Pull request by wbevans**
This pull request primarily focuses on improving the readability and organization of the `README.md` file for the CDF Fabric replicator application. The changes include reformatting of the document structure, addition of more detailed setup instructions, and minor text edits for clarity. It also adds a link to example_config.yml, and removes the text from the Readme itself to eliminate duplicate (and out of date) code.

Key changes include:

Documentation Structure and Formatting:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L3-R67): The document structure has been reorganized with the addition of more subheadings to improve readability and navigation. This includes reformatting of sections such as "Replicator Services", "Local Development with CDF Fabric replicator", "Setting up Data Point Subscriptions", "Environment Variables", "Config YAML", and "Building and Deploying with Docker on AKS" among others.

Detailed Setup Instructions:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L3-R67): More detailed instructions have been added for local development setup with the CDF Fabric replicator, including steps to clone the repository, install Python and Poetry, set up environment variables, and run the application.

Text Edits for Clarity:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L3-R67): Minor text edits have been made throughout the document to improve clarity. This includes changing "Application which utilizes the CDF APIs" to "Application that utilizes the CDF APIs", and "All four services will run concurrently during the run of the CDF Fabric Replicator program" to "All four services will run concurrently during the execution of the CDF Fabric Replicator program".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,121 +1,70 @@
...
+## Replicator Services
 The replicator consists of four services:
-- **Time series replicator** - Copies time series data from CDF to Fabric
-- **Data model replicator** - Copies data model nodes and edges from CDF to Fabric
-- **Event replicator** - Copies event data from CDF to Fabric
-- **Fabric data extractor** - Copies time series, events, and files from Fabric to CDF
+- **Time series replicator** - Copies time series data from CDF to Fabric.
+- **Data model replicator** - Copies data model nodes and edges from CDF to Fabric.
+- **Event replicator** - Copies event data from CDF to Fabric.
+- **Fabric data extractor** - Copies time series, events, and files from Fabric to CDF.
```
Comment (nicholasperkins1992) [inline README.md:10]: A small nit, but should we use the full product names in this document, such as Microsoft Fabric, and Cognite Data Fusion

File: README.md [hunk 2]
```diff
@@ -1,121 +1,70 @@
...
 
-All four services will run concurrently during the run of the CDF Fabric Replicator program.  The services use one state store in CDF's raw storage to maintain checkpoints of when the latest data was copied, so the services can be started and stopped and will be able to pick back up where they left off.
+All four services will run concurrently during the execution of the CDF Fabric Replicator program. The services use one state store in CDF's raw storage to maintain checkpoints of when the latest data was copied, so the services can be started and stopped and will be able to pick back up where they left off.
 
-# Setting up Data Point Subscriptions
+## Local Development with CDF Fabric replicator
+Follow these instructions for doing Local Development with CDF Fabric replicator:
```
Comment (nicholasperkins1992) [inline README.md:15]: Should we also include steps to create a venv, i think that is needed to run locally

File: README.md [hunk 3]
```diff
@@ -1,121 +1,70 @@
...
 - `EXTRACTOR_DATASET_ID`: Specifies the ID of the extractor dataset when the data lands in CDF.
 - `EXTRACTOR_TS_PREFIX`: Specifies the prefix for the extractor timeseries when the data lands in CDF.
 
-## Integration Test Variables
+### Integration Test Variables
 - `TEST_CONFIG_PATH`: Specifies the path to the test configuration file with which test versions of the replicator are configured.
 
-# Config YAML
-The replicator reads its configuration from a YAML file specified in the run command.  You can configure your own YAML file based on the one in `example_config.yaml` in the repo.  That configuration file uses the environment variables in `.env`, the configuration can also be set using hard coded values.
+## Config YAML
```
Comment (nicholasperkins1992) [inline README.md:56]: Wondering if we need a section about the remote config, something to work from or link to https://cognite-extractor-utils.readthedocs-hosted.com/en/latest/remoteconfigs.html

File: README.md [hunk 4]
```diff
@@ -129,7 +78,7 @@ Finally, run the replicator:
 poetry run cdf_fabric_replicator config.yaml
 ```
 
-## Visual Studio Code
+### Visual Studio Code
```
Comment (nicholasperkins1992) [inline README.md:81]: Is this section needed?

File: README.md [hunk 5]
```diff
@@ -1,121 +1,70 @@
 # CDF Fabric replicator
```
Comment (nicholasperkins1992) [inline README.md:1]: Do we need a section explaining how the automatic time series subscription works? That is a bit different from other extractors

File: README.md (hunk 2 above)
Comment (muralidharm) [reply README.md:15]: The venv setup instructions are explained in the poetry docs but might be worth a mention

File: README.md (hunk 4 above)
Comment (muralidharm) [reply README.md:81]: If we're going to have this section might be worth explaining the launch configs that are available (poetry install, poetry run)

File: README.md (hunk 5 above)
Comment (wbevans) [reply README.md:1]: That is already covered in, "Setting up Data Point Subscriptions"

File: README.md (hunk 4 above)
Comment (wbevans) [reply README.md:82]: I do think it's worth having the section. I broke it down as Maya suggested to provide better explanations of the launch configs.

File: README.md [hunk 6]
```diff
@@ -129,7 +78,7 @@ Finally, run the replicator:
 poetry run cdf_fabric_replicator config.yaml
```
[the remark this answers is not in the document]
Comment (wbevans) [reply README.md]: updated to use "<name of config file>" instead of "config.yaml"

File: README.md (hunk 3 above)
Comment (wbevans) [reply README.md:57]: Will cover that in the docs task for "Production", so it will go into another PR.

File: README.md (hunk 2 above)
Comment (wbevans) [reply README.md:15]: Added a note to re-enforce the Poetry venv

**Review by muradsater — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (xtophs) [reply README.md:10]: Spot on. Thanks @wbevans

**Pull request merged**